perun/profile/stats.py

```diff
@@ -344,18 +345,27 @@ def from_values(cls, values: Iterable[float]) -> StatisticalSummary:
         """
         # We assume there aren't too many values so that multiple passes of the list don't matter
         # too much. If this becomes a bottleneck, we can use pandas describe() instead.
-        values = list(values)
-        quantiles = statistics.quantiles(values, n=20, method="inclusive")
-        return cls(
-            float(min(values)),
-            quantiles[2],  # p10
-            quantiles[5],  # p25
-            quantiles[10],  # p50
-            quantiles[15],  # p75
-            quantiles[18],  # p90
-            float(max(values)),
-            statistics.mean(values),
-        )
+        values = list(val for val in values if not math.isnan(val))
+        try:
+            quantiles = statistics.quantiles(values, n=20, method="inclusive")
+            return cls(
+                float(min(values)),
+                quantiles[2],  # p10
+                quantiles[5],  # p25
+                quantiles[10],  # p50
+                quantiles[15],  # p75
+                quantiles[18],  # p90
```

Comment:
These indexes are quite counterintuitive. Is there a reason why `n = 20`? I assume that the quantiles are then mapped from (0, 100) to (0, 20). But it might make sense to refactor it a little.

Reply:
Right, we need to obtain p10, p25, p50, p75, and p90, which means the highest common percentile multiple to compute the desired values is 5 (i.e., it suffices to compute p5, p10, p15, ..., p100). Computing all 100 percentiles is wasteful, as there are generally only few values and the resulting percentiles are interpolated. Any concrete ideas for refactoring this?

Reply:
What about something like
```suggestion
                quantiles[int(0.9 * n)],  # p90
```
It might look more complex, but it is clear this is 90th percentile of the sample `n` and if `n` changes we don't need to recompute it. WDYT?

Reply:
Or we could extract all of this to constants:
```
SAMPLE_SIZE = 20
90_PERCENTILE = int(0.9 * SAMPLE_SIZE)
```